### PR TITLE
Internal: Container - Fix `createContainerFromPreset`. [ED-5713]

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: assets/dev/js/editor/utils/container-helper.js [hunk 1]
```diff
@@ -72,43 +72,130 @@ export class ContainerHelper {
...
+					const sizes = preset.split( '-' ),
+						// Map rounded, user-readable sizes to actual percentages.
+						sizesMap = {
+							33: '33.3333',
+							66: '66.6666',
+						};
+
+					settings = {
+						flex_direction: this.DIRECTION_ROW,
+						flex_wrap: 'wrap',
```
Comment (StyleShit) [inline assets/dev/js/editor/utils/container-helper.js:151]: See here:
https://github.com/elementor/elementor/pull/17024/files#diff-da286180236a3a3fa59034fa28020c17d4333b0f781e51108bb335d0966fe88a

File: assets/dev/js/editor/utils/container-helper.js [hunk 2]
```diff
@@ -72,43 +72,130 @@ export class ContainerHelper {
...
-
-			this.createContainer( {
-				flex_direction: this.DIRECTION_COLUMN,
-				width: {
-					unit: '%',
-					size,
-				},
-			}, parentContainer, { edit: false } );
-		} );
+	static createContainerFromPreset( preset, target = elementor.getPreviewContainer(), options ) {
```
Comment (StyleShit) [inline assets/dev/js/editor/utils/container-helper.js:79]: This function is huge, please extract to smaller functions.

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/utils/container-helper.js (hunk 1 above)
Comment (iNewLegend) [reply assets/dev/js/editor/utils/container-helper.js]: Handled.

File: assets/dev/js/editor/utils/container-helper.js (hunk 2 above)
Comment (iNewLegend) [reply assets/dev/js/editor/utils/container-helper.js:134]: Try now.

File: assets/dev/js/editor/utils/container-helper.js [hunk 3]
```diff
@@ -66,39 +66,39 @@ export class ContainerHelper {
...
 					unit: 'px',
 					size: 0, // Set the gap to 0 to override the default inherited from `Site Settings`.
 				},
-			}, target, options );
+			};
+
+		// Create a parent container to contain all of the sub containers.
+		let parentContainer;
+
+		if ( ! createWrapper ) {
```
Comment (StyleShit) [inline assets/dev/js/editor/utils/container-helper.js:92]: Add a comment that explains that this is used for nested elements.

File: assets/dev/js/editor/utils/container-helper.js [hunk 4]
```diff
@@ -111,12 +111,102 @@ export class ContainerHelper {
 					unit: '%',
 					size,
 				},
+				width_mobile: { // For out-of-the-box responsiveness.
+					unit: '%',
+					size: '100',
+				},
```
Comment (StyleShit) [inline assets/dev/js/editor/utils/container-helper.js:117]: Remove, we don't need that anymore, it's handled by CSS.

File: assets/dev/js/editor/utils/container-helper.js [hunk 5]
```diff
@@ -111,12 +111,102 @@ export class ContainerHelper {
...
+					}
+
+					settings = {
+						width: {
+							unit: '%',
+							size: '50',
+						},
+						width_mobile: {
+							unit: '%',
+							size: '100',
```
Comment (StyleShit) [inline assets/dev/js/editor/utils/container-helper.js:176]: Remove, like above.

File: assets/dev/js/editor/utils/container-helper.js [hunk 6]
```diff
@@ -111,12 +111,102 @@ export class ContainerHelper {
...
+						flex_wrap: 'wrap',
+						flex_gap: {
+							unit: 'px',
+							size: 0, // Set the gap to 0 to override the default inherited from `Site Settings`.
+						},
+					};
+
+					if ( ! createWrapper ) {
+						$e.run( 'document/elements/settings', { container: target, settings } );
+						newContainer = target;
```
Comment (StyleShit) [inline assets/dev/js/editor/utils/container-helper.js:164]: Add empty line:

```suggestion
						$e.run( 'document/elements/settings', { container: target, settings } );
						
						newContainer = target;
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/utils/container-helper.js [hunk 7]
```diff
@@ -66,39 +66,40 @@ export class ContainerHelper {
...
-	 * @param {Container} target - The target container of the newly created Container.
-	 * @param {Object} options - Additional command options.
-	 *
-	 * @return {Container}
-	 */
-	static createContainerFromPreset( preset, target, options = {} ) {
-		const sizes = preset.split( '-' );
-
+	static createContainerFromSizes( sizes, target, options = {} ) {
 		// Map rounded, user-readable sizes to actual percentages.
```
Comment (StyleShit) [inline assets/dev/js/editor/utils/container-helper.js:70]: Move the comment one line down since it refers to the `sizesMap` variable.

**Review by StyleShit — CHANGES_REQUESTED** [review]

**Review by StyleShit — APPROVED** [review]

File: assets/dev/js/editor/utils/container-helper.js [hunk 8]
```diff
@@ -66,39 +66,40 @@ export class ContainerHelper {
...
 
-	/**
-	 * Create a Container element based on a preset.
-	 *
-	 * @param {string} preset - Preset structure of the sub containers (e.g. `33-66-66-33`).
-	 * @param {Container} target - The target container of the newly created Container.
-	 * @param {Object} options - Additional command options.
-	 *
-	 * @return {Container}
-	 */
```
Comment (matipojo) [inline assets/dev/js/editor/utils/container-helper.js:77]: Add back

**Review by matipojo — APPROVED** [review]

File: assets/dev/js/editor/utils/container-helper.js [hunk 9]
```diff
@@ -66,39 +66,40 @@ export class ContainerHelper {
...
-			return sum + parseInt( size );
-		}, 0 );
-
-		const shouldWrap = ( sizesSum > 100 );
-
-		// Create a parent container to contain all of the sub containers.
-		const parentContainer = this.createContainer( {
+	static createContainerFromSizes( sizes, target, options = {} ) {
+		const { createWrapper = true } = options,
+			// Map rounded, user-readable sizes to actual percentages.
```
Comment (matipojo) [inline assets/dev/js/editor/utils/container-helper.js:71]:
```suggestion
			// Map rounded, user-readable sizes to actual percentages.
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Pull request merged**